### ISSUE TITLE
Build Packages for Debian environments.

### ISSUE THREAD

**Issue by timfallmk** [Swordfish90/cool-retro-term#136]
It would be great to have built packages for debian environment, namely Ubuntu. There are lot of build requirements that would make it harder to build yourself.
##

**Comment by tomchiverton** [comment]
This appears to generate a .deb without building anything ?
I created this by replace the debian folder in my recent git clone with the one from your drive. I suspect something is wrong in the packaging but I've never done a source one before :-)

https://launchpad.net/~bugs-launchpad-net-falkensweb/+archive/ubuntu/cool-retro-term/+build/6702457

**Comment by tomchiverton** [comment]
I have succesfully used @JoernSchoenyan debian folder to package to a Ubuntu PPA, and it's install without issues on my Trusty laptop, including all the dependencies.

https://launchpad.net/~bugs-launchpad-net-falkensweb/+archive/ubuntu/cool-retro-term/

**Comment by Swordfish90** [comment]
@tomchiverton Good news. Which version are you currently packaging?

**Comment by Swordfish90** [comment]
If someone is interested in packaging it for Ubuntu 14.10 there is a known issue between the default Qt version and libxext (stdout gets flooded with error messages slowing down the experience). The only solution is to grab the vivid version of libxext.

**Comment by tomchiverton** [comment]
Master as of last night.

**Comment by tomchiverton** [comment]
@Swordfish90 master as of last night

**Comment by Swordfish90** [comment]
@tomchiverton Good. It's better to update the version in the package then. I can still read 0.9, which was a sort of boilerplate version we had before.

**Comment by tomchiverton** [comment]
@Swordfish90 what version should the package be called ? I can easily call it any version regardless of what the internal original source file is called.

Also, I'm not sure what 'vivid version' of libext refers to. In theory we can have a different dependency for 14.10 or statically compile against our own version though, so that's fixable.

**Comment by Swordfish90** [comment]
@tomchiverton This is the ubuntu bug: https://bugs.launchpad.net/ubuntu/+source/xorg/+bug/1369113 . This is related to steam games but the error is the same. This bug was fixed upstream in the following commit: 

http://cgit.freedesktop.org/xorg/lib/libXext/commit/?id=11aad96bd689d54156064d2e81213dc827a689d1

At the moment I've installed the libxext version from vivid on my laptop (http://packages.ubuntu.com/source/vivid/libxext) but I understand this is a suboptimal solution. So if someone has a better idea I'm all ears.

**Comment by tomchiverton** [comment]
@Swordfish90 I'm tempted to publish the package as is for 14.10 then, and note on the PPA web page that you should upgrade that library if you find the performance poor ?

**Comment by Swordfish90** [comment]
@tomchiverton Agreed. I think it's probably the best solution at the moment. We'll see if someone has a better solution in the long term.

**Comment by tomchiverton** [comment]
Done.

**Comment by nodiscc** [comment]
The RFP for a cool-retro-term Debian package is at https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=758238

**Comment by tomchiverton** [comment]
The RFP is over a year old, and needs a lawyer to unpick the licence issues. I think sticking with a PPA is the best plan.

**Comment by nodiscc** [comment]
https://github.com/Swordfish90/cool-retro-term/blob/master/app/qml/fonts/modern-envy-code-r/Read%20Me.txt#L16

**Comment by tomchiverton** [comment]
For instance, yes. @Swordfish90 may have a licence grant we don't know about, for instance.

**Comment by Swordfish90** [comment]
@tomchiverton ... I'm sorry I completely forgot about the licensing issues. I don't have any license grant and I think this needs to be addressed quite soon. To the experts here: getting those licenses would solve all the problems with the Debian packages or we are forced to remove the non-free fonts?

**Comment by nodiscc** [comment]
I think you'd have to completely remove Monaco. I think other non-free fonts would have to be packaged in separate non-free packages, but maybe you could start with a package that provides only free fonts?

You can ask the Debian people by replying to https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=758238

**Comment by tomchiverton** [comment]
My PPA is available for 16.04, so it must be able to build on that platform ?

**Comment by buhtz** [comment]
@tomchiverton Ubuntu 16.04 is far away from Debian **unstable**. It is not the same plattform.

**Comment by tomchiverton** [comment]
Correct. I am unsure of how far the two diverge at the moment. What KDE platform version is unstable ? (K)ubuntu is using 5.18.

**Comment by buhtz** [comment]
I don't use KDE as desktop. Not sure what libs are in the system. You can find details at [Debians Package Tracker](https://tracker.debian.org/). e.g. for [meta-kde](https://tracker.debian.org/pkg/meta-kde).

**Comment by alexmyczko** [comment]
Maybe you want to add an Atari 800 font? www.aiei.ch/atari/

**Comment by alexmyczko** [comment]
Oh and https://github.com/rewtnull/amigafonts (the ttf/ directory)

**Comment by probonopd** [comment]
You could build an [AppImage](http://appimage.org/) on Travis CI and reach users of debian, Ubuntu, CentOS, Fedora, Linux Mint, elementary OS, openSUSE, and many other distributions with just one and the same AppImage.

**Comment by buhtz** [comment]
There are a lot of discussed disadvantages of such image-formats (AppImage, Snappy, etc). It is not recommended. And this Issue is about debian packes and not alternative solutions.

**Comment by probonopd** [comment]
It's not an either-or...

**Comment by jekader** [comment]
As always, pull requests are welcome ;)

**Comment by xaionaro** [comment]
Relates: https://github.com/Swordfish90/cool-retro-term/pull/331

**Comment by buhtz** [comment]
Also see https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=758238

**Comment by probonopd** [comment]
...and https://github.com/Swordfish90/cool-retro-term/pull/336

**Comment by alexmyczko** [comment]
I'm having problems packaging (well running the software) when removing non-free fonts: 
my app/qml/fonts only has modern-hermit
modern-inconsolata
modern-pro-font-win-tweaked
modern-proggy-tiny
modern-terminus

and the others removed, where else do I have to edit that the fonts are gone? Any way you separate the non-free fonts from the building, and have them optional?

**Comment by alexmyczko** [comment]
Guess I have to do the reverse described in: https://github.com/Swordfish90/cool-retro-term/issues/240#issuecomment-180860345 to get the non-free stuff removed from the software. Or maybe replace it with free alternatives...

**Comment by alexmyczko** [comment]
meanwhile http://phd-sid.ethz.ch/debian/cool-retro-term/2020/ a package is ready for official debian inclusion, waiting in NEW queue for review

**Comment by alexmyczko** [comment]
meanwhile if you have debian sid you can apt install cool-retro-term
